src/paths/floyd_warshall.c

```diff
@@ -96,20 +104,34 @@ static igraph_error_t igraph_distances_floyd_warshall_tree(
     for (igraph_integer_t k = 0; k < no_of_nodes; k++) {
         IGRAPH_ALLOW_INTERRUPTION();
 
-        /* resetting no_of_children vector */
-        igraph_vector_int_null(&no_of_children);
-
-        /* constructing the tree out_k (as in the paper) but
-           representing it as the children matrix */
+        /* Count the children of each node in the shortest path tree, assuming that at
+           this point all elements of no_of_children[] are zeros. */
         for (igraph_integer_t v = 0; v < no_of_nodes; v++) {
-            if (k == v) {
-                continue;
-            }
+            if (k == v) continue;
             igraph_integer_t parent = MATRIX(predecessors, k, v);
-            MATRIX(children, parent, VECTOR(no_of_children)[parent]) = v;
             VECTOR(no_of_children)[parent]++;
         }
-        /* constructing dfs-traversal and dfs-skip arrays for the out_k tree */
+
+        igraph_integer_t cumsum = 0;
+        for (igraph_integer_t v = 0; v < no_of_nodes; v++) {
+            VECTOR(children_start)[v] = cumsum;
+            cumsum += VECTOR(no_of_children)[v];
+        }
+        VECTOR(children_start)[no_of_nodes] = cumsum;
```

Comment:
We have an `igraph_vector_int_cumsum()` function - could we use it here?

Reply:
No. I actually added a comment explaining why. I will re-add it.

Given a length-`n` vector `v`, `vector_cumsum()` computes another length-n vector `c` where `c[0] == v[0]` and the last element of `c` is `sum(v)`.

Here we want to start at 0, i.e. `c[0] == 0`, and the length of `c` is `n+1`.